packages/fx-core/src/component/driver/tools/installDriver.ts

```diff
@@ -27,47 +41,58 @@ const helpLink = "https://aka.ms/teamsfx-actions/tools/install";
 
 @Service(ACTION_NAME)
 export class ToolsInstallDriver implements StepDriver {
-  @hooks([addStartAndEndTelemetry(ACTION_NAME, ACTION_NAME)])
   async run(
     args: InstallToolArgs,
     context: DriverContext
   ): Promise<Result<Map<string, string>, FxError>> {
-    const impl = new ToolsInstallDriverImpl(context);
-    return wrapRun(() => impl.run(args));
+    const wrapContext = new WrapDriverContext(context, ACTION_NAME, ACTION_NAME);
+
+    const impl = new ToolsInstallDriverImpl(wrapContext);
+    return (await wrapRun(wrapContext, () => impl.run(args))) as Result<
+      Map<string, string>,
+      FxError
+    >;
+  }
+
+  async execute(args: InstallToolArgs, context: DriverContext): Promise<ExecutionResult> {
+    const wrapContext = new WrapDriverContext(context, ACTION_NAME, ACTION_NAME);
+
+    const impl = new ToolsInstallDriverImpl(wrapContext);
+    return (await wrapRun(wrapContext, () => impl.run(args), true)) as ExecutionResult;
   }
 }
 
 export class ToolsInstallDriverImpl {
-  progressBarName = `Installing tools`;
-  progressBarSteps = 2;
-
-  constructor(private context: DriverContext) {}
+  constructor(private context: WrapDriverContext) {}
 
   async run(args: InstallToolArgs): Promise<Map<string, string>> {
-    // TODO(xiaofhua): prettier output
-    this.context.logProvider.info(`Running '${ACTION_NAME}' driver.`);
     const res = new Map<string, string>();
     this.validateArgs(args);
+
+    this.setArgTelemetry(args);
+    const progressBar = await this.createProgressBar(this.getSteps(args));
+
     if (args.devCert) {
+      await progressBar?.next(ProgressMessages.devCert());
```

Comment:
No need to call `progressBar.start` before?

Reply:
It is in the `createProgressBar` method